src/consensus/block_creation.py

```diff
@@ -115,17 +121,21 @@ def create_foliage(
         prev_block_hash = prev_block.header_hash
 
     solution_program: Optional[SerializedProgram] = None
+    generator_refs: Optional[SerializedProgram] = None
+
     if is_transaction_block:
         aggregate_sig: G2Element = G2Element()
         cost = uint64(0)
 
         if spend_bundle is not None:
-            solution_program = best_solution_program(spend_bundle)
+            solution_program, generator_refs = best_solution_program(spend_bundle)
             aggregate_sig = spend_bundle.aggregated_signature
 
         # Calculate the cost of transactions
         if solution_program is not None:
-            result: CostResult = calculate_cost_of_program(solution_program, constants.CLVM_COST_RATIO_CONSTANT)
+            result: CostResult = calculate_cost_of_program(
+                solution_program, generator_refs, constants.CLVM_COST_RATIO_CONSTANT, False
+            )  # Note: strict_mode == False
```

Comment:
is this where we farm/create a block? If so, it would seem reasonable to use strict mode. @bramcohen thoughts?

Reply:
I think the farming is done here https://github.com/Chia-Network/chia-blockchain/pull/1235/files#diff-86721b338fcff7a1ce541e7d3b414c687023d8ac55a3ebdcc13db46fa0b8b8a1R9

Reply:
Oh I see, it's calculating the cost of the just-farmed block so it can commit that cost.

Reply:
I'd say it's okay to not use strict mode.

Reply:
My reasoning is we don't really care what tricks the generator used, as long as they will validate by others (who are also running in non-strict mode when they validate). Also, someone might want to hack their client to include weird spends that don't work in non-strict, no need to change this code too.